### ISSUE TITLE
[nri-kube-events] number of replicas in the deployment

### ISSUE THREAD

**Issue by gopisaba** [newrelic/helm-charts#47]
**Is your feature request related to a problem? Please describe.**

Is there any specific reason for hardcoding the number of replicas for `nri-kube-events` and `nri-metadata-injection` to `1`?

**Describe the solution you'd like**

It should be at least 2 pods in the deployment for the redundancy.

**Describe alternatives you've considered**

Make it as variable, to allow users to pass from their `values.yaml`

**Comment by douglascamata** [comment]
@gopisaba Increasing the amount of replicas for the event integration will incur in duplicated event data being sent to New Relic, which could make parts of the UI confusing. 

Do you think this can be still useful for you?

**Comment by gopisaba** [comment]
No, not if it makes duplicate entries in NR. I am bit concerned about the redundancy.

The second part of the question is, `nri-metadata-injection` replicas set to `1` as well. Is there any reason for that as well?
I think it is a critical component, if the `metadata-injection` service is down when the new pods come up, the mutation webhook fail to inject the NR environment variables to the new pods and that will be a problem for APM.

**Comment by douglascamata** [comment]
The metadata injection can indeed have the amount of replicas increased without any problem. It's completely stateless. 🤔 
Feel free to open a pull request adding this capability to the chart, if you have time and are comfortable doing it. 💪 

The events integration is a bit complicated to achieve redundancy, as it connects to the k8s api server and streams events. We would need to add a sort of leader election to decide which replica will do the work. This is not currently in our roadmap though. :/
